controllers/knowledgebase_controller.go

```diff
@@ -520,3 +504,7 @@ func (r *KnowledgeBaseReconciler) reconcileDelete(ctx context.Context, log logr.
 		}
 	}
 }
+
+func (r *KnowledgeBaseReconciler) hasHandledPathKey(kb *arcadiav1alpha1.KnowledgeBase, path string) string {
+	return kb.Name + "/" + kb.Namespace + "/" + path
```

Comment:
This should have filegroup's  `source.name` included.  Different filegroup may have same filepath